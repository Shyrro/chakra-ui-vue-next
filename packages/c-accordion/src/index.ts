import { h, defineComponent, PropType } from "vue"
import { chakra, DOMElements } from "@chakra-ui/vue-system"

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: "div",
    },
  },
  setup(props, { slots, attrs }) {
    return () =>
      h(chakra(props.as, { label: "accordion" }), { ...attrs }, slots)
  },
})

export default CAccordion
