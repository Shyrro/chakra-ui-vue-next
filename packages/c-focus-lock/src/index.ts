export * from './c-focus-lock'
