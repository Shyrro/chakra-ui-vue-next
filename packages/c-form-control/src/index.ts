export * from './c-form-control'
export * from './use-form-control'
export * from './c-form-error'