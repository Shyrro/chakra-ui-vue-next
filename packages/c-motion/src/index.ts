export * from './c-motion'
export * from './animate-presence'
export * from './motion-utils'
