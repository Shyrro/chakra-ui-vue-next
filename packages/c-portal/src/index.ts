export { default as CPortal } from "./portal"
