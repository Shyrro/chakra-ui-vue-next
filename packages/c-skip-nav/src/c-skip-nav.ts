/**
 * Hey! Welcome to @chakra-ui/vue-next CSkipNav
 *
 * My component description
 *
 * @see Docs     https://next.vue.chakra-ui.com/c-skip-nav
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue-next/blob/master/packages/c-skip-nav/src/c-skip-nav/c-skip-nav.ts
 * @see WAI-ARIA https://www.w3.org/TR/wai-aria-practices-1.2
 */

import { h, defineComponent } from "vue"
import {
  chakra,
  ComponentWithProps,
  DeepPartial,
  HTMLChakraProps,
  SystemStyleObject,
  useStyleConfig,
} from "@chakra-ui/vue-system"
import { CBox } from "@chakra-ui/vue-next"

const FALLBACK_ID = "chakra-skip-nav"

export interface CSkipNavLinkProps extends HTMLChakraProps<"a"> {}

export const CSkipNavLink: ComponentWithProps<DeepPartial<CSkipNavLinkProps>> =
  defineComponent({
    name: "CSkipNavLink",
    props: {
      id: {
        type: String,
        default: FALLBACK_ID,
      },
    },
    setup(props, { slots, attrs }) {
      function getBaseStyles(styles: any): SystemStyleObject {
        return {
          userSelect: "none",
          border: "0",
          borderRadius: "md",
          fontWeight: "semibold",
          height: "1px",
          width: "1px",
          margin: "-1px",
          padding: "0",
          outline: "0",
          overflow: "hidden",
          position: "absolute",
          clip: "rect(0 0 0 0)",
          ...styles,
          _focus: {
            clip: "auto",
            width: "auto",
            height: "auto",
            boxShadow: "outline",
            padding: "1rem",
            position: "fixed",
            top: "1.5rem",
            insetStart: "1.5rem",
            ...styles["_focusVisible"],
          },
        }
      }

      const styles = useStyleConfig("SkipLink", props)

      return () => {
        return h(
          chakra("a", {
            __css: {
              ...getBaseStyles(styles.value),
            },
          }),
          {
            ...attrs,
            href: `#${props.id}`,
          },
          slots
        )
      }
    },
  })

export interface CSkipNavContentProps extends HTMLChakraProps<"div"> {}

export const CSkipNavContent: ComponentWithProps<
  DeepPartial<CSkipNavContentProps>
> = defineComponent({
  name: "CSkipNavContent",
  props: {
    id: {
      type: String,
      default: FALLBACK_ID,
    },
  },
  setup(props, { attrs, slots }) {
    return () => {
      return h(
        chakra(CBox),
        {
          ...attrs,
          id: props.id,
          tabIndex: "-1",
          "data-testid": FALLBACK_ID,
        },
        slots
      )
    }
  },
})
