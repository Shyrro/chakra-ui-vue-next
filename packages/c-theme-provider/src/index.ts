import {
  h,
  defineComponent,
  Fragment,
  PropType,
  provide,
  inject,
  computed,
} from "vue"
import { Theme } from "@chakra-ui/vue-theme"

const CThemeProvider = defineComponent({
  name: "CThemeProvider",
  props: {
    value: {
      type: [Object] as PropType<Theme>,
      default: () => undefined,
    },
  },
  setup(props, { slots }) {
    const pluginTheme = inject("$chakraTheme")
    const applicationTheme = computed(() => props.value || pluginTheme)
    provide("$chakraTheme", applicationTheme.value)
    return () => h(Fragment, slots.default?.({ $chakraTheme: props.value }))
  },
})

export default CThemeProvider
