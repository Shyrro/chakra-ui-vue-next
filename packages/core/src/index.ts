import { computed, Plugin, Ref, ref, watch } from 'vue'
import defaultTheme, { ColorMode } from '@chakra-ui/vue-theme'
import { toCSSVar, WithCSSVar } from '@chakra-ui/styled-system'
import { chakra, injectGlobal, css } from '@chakra-ui/vue-system'
import internalIcons from './icon.internals'
import { extendTheme, ThemeOverride } from './extend-theme'
import { MergedIcons, parseIcons } from './parse-icons'
import { mode } from '@chakra-ui/vue-theme-tools'

interface ExtendIconsPath {
  path: string
  viewBox?: string
}

interface IconsOptions {
  pack?: 'fa' | 'fe'
  library?: {}
  extend?: Record<string, ExtendIconsPath>
}
export interface ChakraUIVuePluginOptions {
  cssReset?: boolean
  extendTheme?: ThemeOverride
  icons?: IconsOptions
  defaultColorMode?: ColorMode
}

const ChakraUIVuePlugin: Plugin = {
  install(app, options: ChakraUIVuePluginOptions = { cssReset: true }) {
    // 1. Get theme value
    // 2. Parse theme tokens to CSS variables
    // 3. Inject all CSS variables as theme object
    const theme = options.extendTheme || defaultTheme
    const computedTheme = computed<WithCSSVar<ThemeOverride>>(() =>
      toCSSVar(theme)
    )
    injectGlobal({
      ':root': computedTheme.value.__cssVars,
    })

    let libraryIcons = options.icons?.library || {}
    let extendedIcons = options.icons?.extend || {}

    // Initialize colormode
    const colorMode: ColorMode = theme.config?.initialColorMode || 'light'

    // Bind theme to application global properties and provide to application
    app.config.globalProperties.$chakraTheme = computedTheme.value
    app.provide('$chakraTheme', computedTheme.value as ThemeOverride)

    // Provide initial colormode
    app.config.globalProperties.$initialColorMode = colorMode
    const colorModeRef = ref<ColorMode>(colorMode)
    app.provide<Ref<ColorMode>>('$chakraColorMode', colorModeRef)

    libraryIcons = parseIcons(libraryIcons)

    // Merge internal icons and library icons
    const mergedIcons: MergedIcons = {
      ...internalIcons,
      ...libraryIcons,
      ...extendedIcons,
    }

    app.provide('$chakraIcons', mergedIcons)

    // Set color mode property
    app.config.globalProperties.$mode = mode
  },
}

export interface ThemeProviderProps extends ThemeOverride {}
export default ChakraUIVuePlugin
export { extendTheme }

// Export chakra factory function
export { chakra }

/**
 *
 * Component exports
 * ==================
 *
 * Dear contributors,
 *
 * Please keep these exports in Alphabetical order :)
 */

// A
export * from '@chakra-ui/c-accordion'
export * from '@chakra-ui/c-alert'

// B
export * from '@chakra-ui/c-button'

// C
export * from '@chakra-ui/c-color-mode'
export * from '@chakra-ui/c-close-button'
export * from '@chakra-ui/c-code'

// F
export * from '@chakra-ui/c-flex'
export * from '@chakra-ui/c-focus-lock'

// I
export * from '@chakra-ui/c-icon'

// L
export * from '@chakra-ui/vue-layout'

// M
export * from '@chakra-ui/c-modal'
export * from '@chakra-ui/c-motion'

// P
export * from '@chakra-ui/c-popper'
export * from '@chakra-ui/c-portal'

// R
export * from '@chakra-ui/c-reset'

// S
export * from '@chakra-ui/c-spinner'
export * from '@chakra-ui/c-scroll-lock'

// T
export * from '@chakra-ui/c-theme-provider'

// V
export * from '@chakra-ui/c-visually-hidden'

// OTHERS
export * from '@chakra-ui/vue-composables'
export * from '@chakra-ui/vue-a11y'

/**
 *
 * Directives exports
 * ==================
 *
 * Dear contributors,
 *
 * Please keep these exports in Alphabetical order :)
 */
