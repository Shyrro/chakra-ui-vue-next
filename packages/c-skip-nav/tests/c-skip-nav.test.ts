import { render } from '../../test-utils/src'
import { CSkipNav } from '../src'

it('should render properly', () => {
  const { asFragment } = render(CSkipNav)
  expect(asFragment()).toMatchSnapshot()
})