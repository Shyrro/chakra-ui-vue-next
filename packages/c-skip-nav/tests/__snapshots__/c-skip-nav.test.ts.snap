// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSkipNavContent should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-box css-1avyp1d"
    data-testid="chakra-skip-nav"
    id="chakra-skip-nav"
    tabindex="-1"
  />
</DocumentFragment>
`;

exports[`CSkipNavLink should render properly 1`] = `
<DocumentFragment>
  <a
    class="css-fkbol"
    href="#chakra-skip-nav"
  />
</DocumentFragment>
`;
